core/model/src/main/java/org/eclipse/rdf4j/model/impl/DynamicModel.java

```diff
@@ -281,12 +296,32 @@ public void clear() {
 		}
 	}
 
+	@Override
+	public Iterator<Statement> getStatements(Resource subject, IRI predicate, Value object, Resource... contexts) {
+		if (model == null && subject != null && predicate != null && object != null && contexts != null
+				&& contexts.length == 1) {
+			Statement statement = SimpleValueFactory.getInstance()
+					.createStatement(subject, predicate, object, contexts[0]);
+			Statement foundStatement = statements.get(statement);
```

Comment:
Ah. Using a HashMap to be able to match and retrieve the original statement, instead of the newly created copy. Clever.